yetanotherco/aligned_layer
# fix(batcher): send correct responses to each client

Up to now, the batcher responded with the most recent processed batch merkle root to all the connected clients. This was not a correct behaviour because clients should only receive responses from batches where their verification data was included.

To test it:
1. start anvil with `make anvil_start_with_block_time` so that a new block is submitted every 3 seconds.
2. start aggregator and operator.
3. start batcher
4. send a burst of 15 proofs with one of `make batcher_send_groth16_burst`, `make batcher_send_plonk_bls12_381_burst`, `make batcher_send_plonk_bn254_burst` or `make batcher_send_sp1_burst`. You can try this with one or more terminals at the same time. For every message sent by the client, there should also be one response. The responses will be logged as the merkle root of the batch the proof was included in and the hash of the proof. Once all the responses are received, the client will close the connection. If you find that some connection hangs and doesn't close, it may be that the batch was left with only 1 proof. If this happens, try sending an individual proof from another terminal using `make batcher_send_plonk_bn254_task`. The batch should be finalized and a response will arrive for both clients, closing the connection


**NOTE**: For the moment the responses are sent once the batch is processed by the batcher. In a future PR this should be handled in such a way so that it is responded once the full flow on aligned has been done (proofs confirmed in Ethereum)
